### PR TITLE
Fix possible crash when audio channels change

### LINKED ISSUE #16391: Changing audio playback device crashes Godot
Whilst I've not found something specific to my issue, I did notice something that if an audio device was not present to use, Godot will crash, not entirely sure whether these are related. This is my first day using Godot so I'm committed to helping either way. 

**Godot version:**

Godot Engine v3.0.stable.official

**OS/device including version:**

Windows 10, Version 1709 (OS Build 16299.192)

**Issue description:**

Whilst editing a script, my wireless headset died, rather than plug it in I decided to use my monitor's built in speaker to continue listening to my music, upon changing playback device in Windows Sound Godot proceeded to crash to Desktop, I got the "Godot has crashed" but no option to wait, instant crash.

**Steps to reproduce:**
Simply change your audio playback device (whether or not this is only an issue specifically related to wireless headsets or not, I'm unsure) and see whether it crashes.

**Minimal reproduction project:**

No code was involved so not necessary.

### LINKED ISSUE #17453: Game/engine crashes when (un)plugging USB headphones
**Godot version:**
3.0.2

**OS/device including version:**
(OS) Windows 10
(Headphones) Plantronics GameCom 780

**Issue description:**
Plugging/unplugging my USB headphones while in-game or in-engine crashes it.
Bonus: when plugging them, sometimes it won't crash, but the headphones will start making loud painful buzzing noise until I shut down the game/engine/project selector.

It doesn't crash when I plug/unplug standard headphones. Only for USB.

I have searched and found similar issue, but they were all related to Godot 2.x, whereas this happens for 3.x.

### PR DESCRIPTION
**Pull request by marcelofg55**
Fixes #17453 and I think probably fixes #16391 as well.

### CODE REVIEW

File: servers/audio_server.cpp [hunk 1]
```diff
@@ -172,6 +172,13 @@ void AudioServer::_driver_process(int p_frames, int32_t *p_buffer) {
 
 	int todo = p_frames;
 
+	if (channel_count != get_channel_count()) {
+		// Amount of channels changed due to a device change
+		// reinitialize the buses channels and buffers
+		channel_count = get_channel_count();
```
Comment (akien-mga) [inline servers/audio_server.cpp]: This line is not really needed, as you actually do it again at the start of `init_channels_and_buffers()`.

Comment (marcelofg55) [reply servers/audio_server.cpp]: Good catch, just fixed it.

**Comment by akien-mga** [comment]
Thanks!

**Comment by hpvb** [comment]
cherry picked into 3.0.3

**Pull request merged**